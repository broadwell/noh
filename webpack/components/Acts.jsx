import React from "react";

const Acts = () => (
  <div className="acts-container">
    <div className="tl__act" style={{ width: "35%" }}>
      First act (mae ba)
    </div>
    <div className="tl__act" style={{ width: "65%" }}>
      Second act (nochi ba)
    </div>
  </div>
);

export default Acts;
