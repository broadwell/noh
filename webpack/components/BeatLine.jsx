/* eslint-disable react/no-array-index-key */

import React, { Component } from 'react';
import PropTypes from 'prop-types';

class BeatLine extends Component {
  constructor (props) {
    super(props)
    this.state = {
      value: this.props.value
    };
  }

  onChangeCell(event, index) {
    const { value } = this.state;
    value[index] = event.target.value;
    this.setState({ value });
  }

  render() {
    const cells = Array(this.props.grid).fill(0).map((v, i) => {
      let cellContent = '';
      if (this.props.cellType === 'text') {
        cellContent = (
          <input
            maxLength={ 2 }
            onChange={ (event) => this.onChangeCell(event, i) }
            value={ this.state.value[i] }
          />
        );
      } else {
        cellContent = <span />;
      }
      return (
        <div className="cell" key={ i }>
          { cellContent }
        </div>
      );
    });
    return (
      <div className="beat-line">
        <div className="label">{ this.props.label }</div>
        <div className="cell-container">
          { cells }
        </div>
      </div>
    );
  }
}

BeatLine.propTypes = {
  value: PropTypes.arrayOf(PropTypes.string),
  label: PropTypes.string.isRequired,
  grid: PropTypes.number,
  cellType: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.element
  ])
}

BeatLine.defaultProps = {
  grid: 16,
  value: [],
  cellType: 'text'
}

export const Unwrapped = BeatLine;
export default BeatLine;
