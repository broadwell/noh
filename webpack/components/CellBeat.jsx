import React from "react";
import PropTypes from "prop-types";

const CellBeat = props => (
  <div className="cell cell--1 cell--beat">{props.beatText}</div>
);

CellBeat.propTypes = {
  beatText: PropTypes.string.isRequired
};

export default CellBeat;
