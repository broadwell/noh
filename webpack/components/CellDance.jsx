import React from "react";
import PropTypes from "prop-types";

const CellDance = props => (
  <div className={`cell cell--${props.length} cell--dance`}>
    <span data-tooltip={props.text} className="truncate">
      {props.text}
    </span>
  </div>
);

CellDance.propTypes = {
  text: PropTypes.string.isRequired,
  length: PropTypes.number.isRequired
};

export default CellDance;
