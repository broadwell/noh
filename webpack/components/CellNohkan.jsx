import React from "react";
import PropTypes from "prop-types";

const CellNohkan = props => (
  <div className={`cell cell--${props.length} cell--nohkan`}>
    <span data-tooltip={props.text} className="truncate">
      {props.text}
    </span>
  </div>
);

CellNohkan.propTypes = {
  text: PropTypes.string.isRequired,
  length: PropTypes.number.isRequired
};

export default CellNohkan;
