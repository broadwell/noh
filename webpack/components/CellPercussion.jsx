import React, { Component } from "react";
import PropTypes from "prop-types";

class CellPercussion extends Component {
  render() {
    const cellPercussionClass =
      this.props.text.length > 0 ? "cell--percussion" : "";

    return (
      <div className={`cell cell--${this.props.length} ${cellPercussionClass}`}>
        <span className="truncate">{this.props.text}</span>
      </div>
    );
  }
}

CellPercussion.propTypes = {
  text: PropTypes.string.isRequired,
  length: PropTypes.number.isRequired
};

export default CellPercussion;
