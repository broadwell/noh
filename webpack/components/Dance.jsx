import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import data from '../data/kiri-data.json';
import SimpleVideo from './SimpleVideo';

class Dance extends Component {
  currentDance() {
    const chunks = data.kiri.dance;
    const dance = { name: '', url: '' };
    for (let i = 0; i < chunks.length; i += 1) {
      if (
        this.props.currentTime >= chunks[i].timeStart &&
        this.props.currentTime < chunks[i].timeEnd
      ) {
        dance.name = chunks[i].name;
        dance.url = chunks[i].url;
      }
    }
    return dance;
  }

  render() {
    let danceVideo = null;
    const dance = this.currentDance();
    if (dance.name !== '') {
      danceVideo = <SimpleVideo src={dance.url} />;
    }
    return (
      <div className="dance-style">
        <h2>Dance Style: {this.currentDance().name}</h2>
        {danceVideo}
      </div>
    );
  }
}

Dance.propTypes = {
  currentTime: PropTypes.number
};

Dance.defaultProps = {
  currentTime: 0
};

const mapStateToProps = state => ({ currentTime: state.currentTime });

export const Unwrapped = Dance;
export default connect(mapStateToProps)(Dance);
