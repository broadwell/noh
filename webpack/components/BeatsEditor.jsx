import React, { Component } from 'react';
import PropTypes from 'prop-types';
import FileDownload from 'js-file-download';
import BeatLine from './BeatLine';

class BeatsEditor extends Component {

  constructor(props) {
    super(props);
    this.state = {
      filename: this.props.filename,
      beats: this.props.beats,
      syllable_text: this.props.syllable_text,
      syllable_number: this.props.syllable_number,
      vocal: this.props.vocal,
      nokan: this.props.nokan,
      dance: this.props.dance
    };
    this.onClick = this.onClick.bind(this);
    this.onChange = this.onChange.bind(this);
  }

  onClick(event) {
    const target = event.target;
    const value = target.value;
    const name = target.name;
    this.setState({
      [name]: value
    });
    FileDownload(JSON.stringify(this.state), this.state.filename);
  }

  onChange(event) {
    this.setState({
      filename: event.target.value
    })
  }

  render() {
    return (
      <div>
        <BeatLine
          label="Beats"
          name="beats"
          grid={this.props.grid}
          cellType="text"
          value={ this.state.beats }
        />
        <BeatLine
          label="Syllable text"
          name="syllable_text"
          grid={this.props.grid}
          cellType="text"
          value={ this.state.syllable_text }
        />
        <BeatLine
          label="Syllable number"
          name="syllable_number"
          grid={this.props.grid}
          cellType="text"
          value={ this.state.syllable_number }
        />
        <BeatLine
          label="Vocal range"
          name="vocal"
          grid={this.props.grid}
          cellType="click"
        />
        <BeatLine
          label="Nokan"
          name="nokan"
          grid={this.props.grid}
          cellType="click"
        />
        <BeatLine
          label="Dance"
          name="dance"
          grid={this.props.grid}
          cellType="click"
        />
        <input type="button" value="Download" onClick={ this.onClick }/>
        <span>as</span>
        <input
          name="filename"
          type="text"
          value={ this.state.filename }
          onChange={ this.onChange }
        />
      </div>
    );
  }
}

BeatsEditor.propTypes = {
  grid: PropTypes.number,
  filename: PropTypes.string,
  beats: PropTypes.array,
  syllable_text: PropTypes.array,
  syllable_number: PropTypes.array,
  vocal: PropTypes.array,
  nokan: PropTypes.array,
  dance: PropTypes.array
}

BeatsEditor.defaultProps = {
  grid: 16,
  filename: 'beats.json',
  beats: [],
  syllable_text: [] ,
  syllable_number: [],
  vocal: [],
  nokan: [],
  dance: []
}


export const Unwrapped = BeatsEditor;
export default BeatsEditor;
