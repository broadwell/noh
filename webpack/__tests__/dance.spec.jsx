import React from 'react';
import { shallow } from 'enzyme';
import { Unwrapped as UnwrappedDance } from '../components/Dance';

describe('<Dance>', () => {
  it('renders as expected with no currentTime', () => {
    const component = shallow(<UnwrappedDance />);
    expect(component).toMatchSnapshot();
  });

  it('renders as expected with currentTime', () => {
    const component = shallow(<UnwrappedDance currentTime={30} />);
    expect(component).toMatchSnapshot();
  });
});
