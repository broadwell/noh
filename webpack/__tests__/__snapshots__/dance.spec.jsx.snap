// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dance> renders as expected with currentTime 1`] = `
<div
  className="dance-style"
>
  <h2>
    Dance Style: 
    Ageha
  </h2>
  <SimpleVideo
    src="https://www.dropbox.com/s/e5hv7tovg11qp3d/ageha_Front.mov?dl=1"
  />
</div>
`;

exports[`<Dance> renders as expected with no currentTime 1`] = `
<div
  className="dance-style"
>
  <h2>
    Dance Style: 
  </h2>
</div>
`;
